infra/build_specified_commit.py

```diff
@@ -22,75 +22,81 @@
     commit: The commit SHA to build the fuzzers at
-    local_store_path: The full file path of a place where a temp git repo is stored
+    build_repo_manager: The OSS-Fuzz project's repo manager to be built at
     engine: The fuzzing engine to be used
     sanitizer: The fuzzing sanitizer to be used
     architecture: The system architiecture to be used for fuzzing
 
   Returns:
     0 on successful build 1 on failure
   """
-  if not old_repo_manager:
-    inferred_url, repo_name = detect_main_repo_from_docker(project_name, commit)
-    old_repo_manager = repo_manager.RepoManager(
-        inferred_url, local_store_path, repo_name=repo_name)
-  old_repo_manager.checkout_commit(commit)
+  build_repo_manager.checkout_commit(commit)
   return helper.build_fuzzers_impl(
       project_name=project_name,
       clean=True,
       engine=engine,
       sanitizer=sanitizer,
       architecture=architecture,
       env_to_add=None,
-      source_path=old_repo_manager.repo_dir,
-      mount_location=os.path.join('/src', old_repo_manager.repo_name))
+      source_path=build_repo_manager.repo_dir,
+      mount_location=os.path.join('/src', build_repo_manager.repo_name))
 
 
-def detect_main_repo_from_docker(project_name, example_commit, src_dir='/src'):
+def detect_main_repo(project_name, repo_name=None, commit=None, src_dir='/src'):
   """Checks a docker image for the main repo of an OSS-Fuzz project.
 
+  Note: The default is to use the repo name to detect the main repo
+
   Args:
-    project_name: The name of the OSS-Fuzz project
-    example_commit: An associated commit SHA
+    project_name: The name of the oss-fuzz project
+    repo_name: The name of the main repo in an OSS-Fuzz project
+    commit: A commit SHA that is associated with the main repo
     src_dir: The location of the projects source on the docker image
 
   Returns:
     The repo's origin, the repo's name
   """
+  if not repo_name and not commit:
+    print('Error can not detect main repo without a repo_name or a commit')
+    return None, None
+
+  if repo_name:
+    helper.build_image_impl('base-builder')
   helper.build_image_impl(project_name)
   docker_image_name = 'gcr.io/oss-fuzz/' + project_name
   command_to_run = [
-      'docker', 'run', '--rm', '-i', '-t', docker_image_name, 'python3',
-      os.path.join(src_dir, 'detect_repo.py'), '--src_dir', src_dir,
-      '--example_commit', example_commit
-  ]
+      'docker', 'run', '--rm', '-t', docker_image_name, 'python3',
+      os.path.join(src_dir, 'detect_repo.py'), '--src_dir', src_dir]
+  if repo_name:
+    command_to_run.extend(['--repo_name', repo_name])
+  else:
+    command_to_run.extend(['--example_commit', commit])
   out, _ = execute(command_to_run)
-
   match = re.search(r'\bDetected repo: ([^ ]+) ([^ ]+)', out.rstrip())
   if match and match.group(1) and match.group(2):
-    return match.group(1), match.group(2).rstrip()
+    return match.group(1), match.group(2)
   return None, None
 
 
+
```

Comment:
format and lint please. These seem unnecessary.

Reply:
Done.